tsconfig-base.json

```diff
@@ -0,0 +1,27 @@
+{
+  "compilerOptions": {
+    "moduleResolution": "node",
```

Comment:
Not super familiar with tsconfig options but it seems strange to have `"moduleResolution": "node"` in the base. Is this true for esm modules aswell?

Reply:
We could use `node16` or `nodenext` for esm but that requires typescript 4.7

Reply:
ok i'm no opinionated about it just wanted to flag